src/llm/apis/openai_completions.cpp

```diff
@@ -75,32 +76,139 @@ absl::Status OpenAIChatCompletionsHandler::parseCompletionsPart() {
+                            if (!entry.HasMember("image_url") || !entry["image_url"].IsObject()) {
+                                return absl::InvalidArgumentError("Invalid message structure - content image_url missing");
+                            }
+                            auto imageUrl = entry["image_url"].GetObject();
+                            if (!imageUrl.HasMember("url") || !imageUrl["url"].IsString()) {
+                                return absl::InvalidArgumentError("Invalid message structure - image_url does not have url field");
+                            }
+                            std::string url = imageUrl["url"].GetString();
+                            std::string pattern = ";base64,";
+                            std::size_t found = url.find(pattern);
+                            size_t offset = found + pattern.length();
+                            if (found == std::string::npos) {
+                                offset = 0;
+                            }
+                            url.erase(0, offset);
+                            std::string decoded;
+                            if (!absl::Base64Unescape(url, &decoded)) {
+                                return absl::InvalidArgumentError("Invalid base64 string in request");
+                            }
+                            std::vector<unsigned char> data(decoded.begin(), decoded.end());
+                            cv::Mat dataMat(data, true);
+                            cv::Mat image;
+                            try {
+                                image = cv::imdecode(dataMat, cv::IMREAD_UNCHANGED);
+                            } catch (const cv::Exception& e) {
+                                return absl::InvalidArgumentError("Error during string to mat conversion");
+                            }
+                            std::vector<size_t> shape;
+                            shape.push_back(image.rows);
+                            shape.push_back(image.cols);
+                            shape.push_back(image.channels());
+                            ov::Tensor tensor(getOvTypeFromMatType(image.depth()), shape);
+                            if (image.total() * image.elemSize() != tensor.get_size()) {
+                                return absl::InvalidArgumentError("Image size invalid");
+                            }
+                            memcpy((char*)tensor.data(), (char*)image.data, image.total() * image.elemSize());
+                            request.images.push_back(tensor);
+                        } else {
+                            return absl::InvalidArgumentError("Unsupported content type");
+                        }
+                    }
+                    member->value = contentText;
+                } else {
+                    return absl::InvalidArgumentError("Invalid message structure - content should be string or array");
+                }
+            }
         }
     }
+    if (jsonChanged) {
```

Comment:
I'm not sure if this trick really speeds in the processing since you need to make a copy anyway: `request.processedJson = buffer.GetString();`